File: ArraysAndHashing/python/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        seen = {}
        for idx, val in enumerate(nums):
            res = target-val
            if res in seen:
                return [idx, seen[res]]
            seen[val] = idx
        return

File: ArraysAndHashing/js/1_Two_Sum.js
var twoSum = function(nums, target) {
    const seen = new Map();
    for(let i = 0; i < nums.length; i++){
        if(seen.has(target - nums[i])){
            return [seen.get(target - nums[i]), i]
        }
        seen.set(nums[i], i)
    }
    return [0,0]
};